runtime/bin/stdio_win.cc

```diff
@@ -42,8 +54,8 @@ bool Stdin::GetEchoMode(intptr_t fd, bool* enabled) {
   return true;
 }
 
-bool Stdin::SetEchoMode(intptr_t fd, bool enabled) {
-  HANDLE h = GetStdHandle(STD_INPUT_HANDLE);
+bool Stdin::SetEchoMode(intptr_t fd, bool enabled, int stdioMode) {
+  HANDLE h = GetInpStdHandle(stdioMode);
```

Comment:
Immediate question that comes up is whether updating console mode (via CONIN$) will affect behavior of redirected stdin that dart app deals with. Basically if dart app reads from stdin which might originate from console, might originate from somewhere else, updating actual console mode might give developer false sense of successfully completed operation.

Reply:
> The standard handles of a process may have been redirected by a call to SetStdHandle, in which case GetStdHandle will return the redirected handle. If the standard handles have been redirected, you can specify the CONIN$ value in a call to the CreateFile function to get a handle to a console's input buffer. Similarly, you can specify the CONOUT$ value to get a handle to the console's active screen buffer.

As by this from [here](https://learn.microsoft.com/en-us/windows/console/setstdhandle) i believed by CreatFule with Conin$ we would get the actual handle which is the effective handle controlling the console, but ur question eagers me to disbelief this i will test this and will let u know, thanks for reviewing.